pkg/datapath/loader/base.go

```diff
@@ -511,3 +513,63 @@ func (l *Loader) Reinitialize(ctx context.Context, o datapath.BaseProgramOwner,
 
 	return nil
 }
+
+// OnUpdateCiliumNode is only called if running with IPAM ENI and IPsec. In
+// that case, we need to rerun reinitializeIPSec whenever new ENIs are added to
+// the local node, to attach our BPF program for IPsec decryption to the new
+// interfaces.
+func (l *Loader) OnUpdateCiliumNode(oldNode, newNode *ciliumv2.CiliumNode, _ *lock.StoppableWaitGroup) error {
+	if !k8s.IsLocalCiliumNode(newNode) {
+		return nil
+	}
+
+	log.Debug("Checking for new ENIs on local node")
+
+	newENIs := make([]string, 0)
+	for newENI := range newNode.Status.ENI.ENIs {
+		existed := false
+		for oldENI := range oldNode.Status.ENI.ENIs {
+			if newENI == oldENI {
+				existed = true
+				break
+			}
+		}
+		if !existed {
+			newENIs = append(newENIs, newENI)
+		}
+	}
+
+	if len(newENIs) == 0 {
+		return nil
+	}
+
+	// ... and we can then trigger loading of BPF programs to those interfaces.
+	log.WithField(logfields.NewENIs, newENIs).Info("New ENIs detected")
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// If we have any new ENIs, we first need to wait for the corresponding
+	// interfaces to show up...
+	missingENIByMac, err := ipam.WaitForENIInterfaces(oldNode, newNode)
```

Comment:
This will block any CiliumNode events from being processed in the agent. We should move this into a go routine, a [job group](https://docs.cilium.io/en/latest/contributing/development/hive/#job-groups) or good old fashioned one-shot [controller](https://pkg.go.dev/github.com/cilium/cilium@v1.13.2/pkg/controller)